### ISSUE TITLE
teams-for-linux can't open meeting link from command line

### ISSUE THREAD

**Issue by bigiulio** [IsmaelMartinez/teams-for-linux#318]
**Describe the bug**
Since yesterday (or the day before yesterday) after opening links with `teams-for-linux --url {URL}` (see the URL below) the content inside teams-for-linux window flashes for a few times and then teams-for-linux ends with:

`The resource you are looking for has been removed, had its name changed, or is temporarily unavailable.`

**To Reproduce**

	$ teams-for-linux --url https://teams.microsoft.com/l/meetup-join/19%3ameeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy%40thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d
	configPath = ~/.config/teams-for-linux
	Failed to get the config file, using default values
	configFile = {}
	Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"
	processArgs [ '/opt/teams-for-linux/teams-for-linux',
	  '--url',
	  'https://teams.microsoft.com/l/meetup-join/19%3ameeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy%40thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d' ]
	meetup-join argument received with https protocol
	second-instance started
	processArgs [ '/opt/teams-for-linux/teams-for-linux',
	  'msteams:/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d&anon=true&deeplinkId=698d521e-ad93-42e1-2233-e9a0e6ef91b2&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com' ]
	meetup-join argument received with msteams protocol
	second-instance started
	processArgs [ '/opt/teams-for-linux/teams-for-linux',
	  'msteams:/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d&anon=true&deeplinkId=698d521e-ad93-42e1-2233-e9a0e6ef91b2&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com&deeplinkId=7324dd94-754d-49ab-9e94-23faaa51fbc0&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com' ]
	meetup-join argument received with msteams protocol
	second-instance started
	processArgs [ '/opt/teams-for-linux/teams-for-linux',
	  'msteams:/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d&anon=true&deeplinkId=698d521e-ad93-42e1-2233-e9a0e6ef91b2&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com&deeplinkId=7324dd94-754d-49ab-9e94-23faaa51fbc0&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com&deeplinkId=483346ad-a62e-4ac4-b480-5605e929cc0f&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com' ]
	meetup-join argument received with msteams protocol
	second-instance started
	processArgs [ '/opt/teams-for-linux/teams-for-linux',
	  'msteams:/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d&anon=true&deeplinkId=698d521e-ad93-42e1-2233-e9a0e6ef91b2&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com&deeplinkId=7324dd94-754d-49ab-9e94-23faaa51fbc0&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com&deeplinkId=483346ad-a62e-4ac4-b480-5605e929cc0f&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com&deeplinkId=5964824e-f264-4cfa-79ea-23e101b96536&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=teams.microsoft.com' ]
	meetup-join argument received with msteams protocol
	window close
	^Cbefore-quit
	window close
	^C
	$ 

Basically, what happens is that the content inside teams-for-linux window flashes for a few times and then teams-for-linux ends with:

`The resource you are looking for has been removed, had its name changed, or is temporarily unavailable.`
**Expected behavior**
A screen saying something like "Waiting for someone to allow you to get in"

**Screenshots**
See attached screencast

**Desktop (please complete the following information):**
teams-for-linux_1.0.0_amd64.deb

**Additional context**
The same happens with vanilla live cd Debian Buster distribution (and with `teams-for-linux_1.0.0_amd64.deb` as well of course) while on Windows clients the link works fine :(
No issues doing calls and all the other things teams-for-linux allows you to do when not opening a link

[Screencast from 02-07-2020 11:16:41 PM.zip](https://github.com/IsmaelMartinez/teams-for-linux/files/4173137/Screencast.from.02-07-2020.11.16.41.PM.zip)

**Comment by IsmaelMartinez** [comment]
Can you Try one of the  MSTeams links?
msteams:/l/meetup-join/19:meeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy@thread.v2
/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d&anon=true&deeplinkId=698d521e-ad93-42e1-2233-e9a0e6ef91b2&launchAgent=join_launcher&type=meetup-join&directDl=true&msLaunch=true&enableMobilePage=true&fqdn=
teams.microsoft.com

**Comment by bigiulio** [comment]
Not sure what link I should try.

This is what I tried:

```
$ teams-for-linux --url msteams://teams.microsoft.com/l/meetup-join/19%3ameeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy%40thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d   
configPath = ~/.config/teams-for-linux
Failed to get the config file, using default values
configFile = {}
Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"
processArgs [ '/opt/teams-for-linux/teams-for-linux',
  '--url',
  'msteams://teams.microsoft.com/l/meetup-join/19%3ameeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy%40thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d' ]
second-instance started
processArgs [ '/opt/teams-for-linux/teams-for-linux',
  'msteams://teams.microsoft.com/l/meetup-join/19%3ameeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy%40thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d' ]


window close
^Cbefore-quit
window close
^C
$ 
```

Otherwise, can you please rewrite the URL I should use?

**Comment by IsmaelMartinez** [comment]
Sorry, not much of an internet connection during this weekend and next
week. Try pasting the msteams: link that I pasted before in your browser.
If you had installed the Debian version it will try to open the link in
teams-for-linux.

Otherwise, try a version before 1.0.0 but it's probably something that ms
have changed and that now has broken.

Try with another meeting created by yourself just in case the link you
got doesn't work anymore (they cancelled the meeting).

Hope helps

**Comment by bigiulio** [comment]
I just tried the following (which is basically the link you suggested but URL encoded) getting again a completely blank page:

	teams-for-linux --url msteams://teams.microsoft.com/l/meetup-join/19%3ameeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy%40thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d%26anon=true%26deeplinkId=698d521e-ad93-42e1-2233-e9a0e6ef91b2%26launchAgent=join_launcher%26type=meetup-join%26directDl=true%26msLaunch=true%26enableMobilePage=true%26fqdn=teams.microsoft.com

To sum up, please note that, if you use the URL I've provided in my very first comment with teams-for-linux (either triggerd by Chrome or command line) then you'll be getting the the same (wrong) behaviour __but__ if you use the same link and join with Chrome, then you'll be asked to wait for someone to let you join in (expected behaviour).

**Comment by IsmaelMartinez** [comment]
Hi, just to confirm, you say this use to work in teams-for-linux a few
days ago?

**Comment by bigiulio** [comment]
Yes, I can't be more accurate, but even if being little bit conservative, I can say that for sure it was fully functional till Fri, Jan 24th

**Comment by IsmaelMartinez** [comment]
Ok, is a bit of a pain but the only way I can see it working is by copying the meeting link into a teams channel and then click on the link. Then it works... looks like MS have changed something on their end. The link like:

https://teams.microsoft.com/l/meetup-join/19%3ameeting_stuf.....

**Comment by bigiulio** [comment]
I'm unsure of what you mean.

I've created a private Channel and copy pasted my Teams URL in the Channel posts. But I'm not getting a clickable URL. What I'm doing wrong?


Thanks

**Comment by IsmaelMartinez** [comment]
What I did was:

- Create a new channel (well, I just used one that I got for test and only me is in it)

- Copy paste the following link to the channel (and "send")
https://teams.microsoft.com/l/meetup-join/19%3ameeting_YzcyNTYxM2EtYzI0OS00OTA0LWIxZjEtMGI4Mjk0NmU5YjMy%40thread.v2/0?context=%7b%22Tid%22%3a%2205f73a75-999a-4b6f-83c1-8a9511c3f90d%22%2c%22Oid%22%3a%221503686b-c143-4d30-ab15-a4acaea621c7%22%7d

- You should get something like this.
![image](https://user-images.githubusercontent.com/430992/74153086-a8481600-4c07-11ea-8cd7-43ed3c619c30.png)

Just click on the link with your mouse. It works like this:
![318-teams-for-linux](https://user-images.githubusercontent.com/430992/74153158-dfb6c280-4c07-11ea-8d5a-e9c95646e34f.gif)

by the way, make sure you don't use that link again as this is public...

**Comment by IsmaelMartinez** [comment]
I think this one and #322 are more or less the same. MS have changed something in the way they open the links and now they fail when launch from the outside. Having a look but slowly as I got not much time.

**Comment by IsmaelMartinez** [comment]
I am looking at this and have identified where the issue is but it's a bit of a precision work... Meaning, adding the fix is breaking other functionality... But I am working on it.

**Comment by bigiulio** [comment]
I'm running the application from git source.

As soon as you're done, let me know and I'll do a pull and a try on it.

**Comment by IsmaelMartinez** [comment]
Hi, There is a PR open for this. Codebuild is compliaining about me adding some console statements but the fix should be there. 
When running from yarn you will get a "can't find handler for msteams" error message pop up but will still work. That works fine once packaged.

**Comment by bigiulio** [comment]
I would say all fine and good. For a couple of days I tried starting Teams up like this:
`$ yarn start --url {URL}`
with no issues at all (neither "can't find handler for msteams").

Thanks

**Comment by IsmaelMartinez** [comment]
Nice to hear that. Snap edge is using that. I will see when I can get that released as soon as I can. I am taking the advantage of moving to GitHub actions from Travis as they Travis is slowing me down. I will ping you once is done and close the issue. Apologies but busy life!!

**Comment by IsmaelMartinez** [comment]
Also, I am fairly sure you don't need the --url. You can just do `teams-for-linux {URL}`

**Comment by josefschabasser** [comment]
As mentioned in #341 I moved to `snapd` and installed version 1.0.4.
Now this happens:
* I click a meeting link
* if teams-for-linux has an open window it is moved to foreground, if there is no window (e.g. minimized to tray) nothing happens

Output when I try to run it from the command line:
```
$ teams-for-linux {URL}
configPath = /home/{USER}/snap/teams-for-linux/98/.config/teams-for-linux
Failed to get the config file, using default values
configFile = {}
Error org.freedesktop.DBus.Error.Failed: cannot use setting "default-url-scheme-handler": not allowed
App already running
```

**Comment by IsmaelMartinez** [comment]
I don’t think snap allows for this (issue in it). This is duplicated with https://github.com/IsmaelMartinez/teams-for-linux/issues/175#issue-468380730

Use another package type (deb or rpm and/or wait for snaps to fix the issue (https://github.com/snapcore/snapd/pull/7129)

**Comment by IsmaelMartinez** [comment]
Sorry, closed by confusion with @josefschabasser comment. Hope to have a release at some point... just don’t have time to pack stuff up

**Comment by IsmaelMartinez** [comment]
Pre release ready in here https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.0.4

Can someone test it? I have been running it for 11 days with no issues but there are quite a few changes... internally

**Comment by bigiulio** [comment]
Started using v1.0.4 tag.

Will update in 7 days....

**Comment by IsmaelMartinez** [comment]
Should I assume no news good news @bigiulio ?

**Comment by bigiulio** [comment]
You should assume, no time ;)

The only nasty bug is, when you're in a meeting, and you click Email icon from Contacts, meeting is dropped. This happened at all times in my conf.

Beside that, it's all stable enough

**Comment by IsmaelMartinez** [comment]
Is that something that was happening before? I will probably then try to release properly on this week. Thanks!

**Comment by bigiulio** [comment]
Mmmmh, can't say for sure, sorry

**Comment by IsmaelMartinez** [comment]
released as 1.0.5. Closing
